src/SourceGenerators/Uno.UI.Tasks/ResourcesGenerator/ResourcesGenerationTask.cs

```diff
@@ -49,15 +50,27 @@ public override bool Execute()
 
 			try
 			{
-				GeneratedFiles = Directory
-					.EnumerateDirectories(ResourcesDirectory)
-					.Select(directory =>
+				GeneratedFiles = Resources
+					// TODO: Add support for other resources file names
+					.Where(resource => resource.ItemSpec?.EndsWith("Resources.resw") ?? false)
+					// TODO: Merge duplicates (based on file name and qualifiers)
+					.Select(resource =>
 					{
-						var language = new DirectoryInfo(directory).Name;
+						this.Log().Info("Resources file found : {0}".InvariantCultureFormat(resource.ItemSpec));
+
+						var resourceCandidate = ResourceCandidate.Parse(resource.ItemSpec, resource.ItemSpec);
+
+						var language = resourceCandidate.GetQualifierValue("language");
+						if (language == null)
+						{
+							// TODO: Add support for resources without a language qualifier
+							this.Log().Info("No language found, resources ignored".InvariantCultureFormat(language));
```

Comment:
Git seems to've swallowed my comment... The format string doesn't look right here and it should be logging resourceCandidate, not language (which is null)

Reply:
Fixed.